### PR TITLE
Onyx LSH

### PR DESCRIPTION
**Pull request by punit-naik**
# Status
Please select the status of this PR-

- [x] Ready to review and merge
- [ ] Commits pending
- [ ] On Hold

# Committer Checklist
Overview of the changes proposed in this PR.

## Changes Proposed
Added Clojure - Onyx Implementations

## Test Results
N/A

## Related PRs
N/A

## Documentation
N/A

# Reviewer Checklist
Please validate and mark the readiness of this PR based on these validations-

## Pre-requisites
- [x] Follows [Coding Style](https://formcept.atlassian.net/wiki/display/DEV/Developer+Handbook#DeveloperHandbook-CodingStyleGuide)
- [x] Follows [Development Workflow](https://formcept.atlassian.net/wiki/display/DEV/Developer+Handbook#DeveloperHandbook-DevelopmentWorkflow) and targets the right branch
- [x] Follows [Commit Message Guidelines](https://formcept.atlassian.net/wiki/display/DEV/Developer+Handbook#DeveloperHandbook-Commit)
- [ ] Has an associated JIRA ID and the same has been mentioned in the PR title

## Change Checklist
- [ ] PR contains only the changes expected in the mentioned JIRA ticket(s)
- [ ] Test Cases have been added
- [ ] Changes have been documented in code and/or Confluence

### CODE REVIEW

**Comment by punit-naik** [comment]
@anujsrc How does this look?

File: clojure/build.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+echo -e "Building Clojure Implementations...\n"
+
+##################
+# ONYX
+
+cd onyx-impl
```
Comment (anujsrc) [inline clojure/build.sh]: @punit-naik - lets keep the top-level folder name as the platform name. So, change this to ``onyx``.

File: clojure/onyx-impl/README.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+ML-Onyx
```
Comment (anujsrc) [inline clojure/onyx-impl/README.md]: Change to just ``Onyx``

File: clojure/onyx-impl/build.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+echo -e "Building Onyx Implementations..."
```
Comment (anujsrc) [inline clojure/onyx-impl/build.sh]: Just say ``Onyx``

File: clojure/onyx-impl/locality-sensitive-hashing/README.md [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# lsh
```
Comment (anujsrc) [inline clojure/onyx-impl/locality-sensitive-hashing/README.md]: Change to full-form- ``Locality Sensitive Hashing (LSH)``

File: clojure/onyx-impl/onyx-utils/README.md [hunk 5]
```diff
@@ -0,0 +1,4 @@
+# onyx-utils
```
Comment (anujsrc) [inline clojure/onyx-impl/onyx-utils/README.md]: Change to ``Onyx Utilities``

File: clojure/onyx-impl/locality-sensitive-hashing/src/org/formcept/onyx/lsh_onyx/core.clj [hunk 6]
```diff
@@ -0,0 +1,252 @@
+(ns org.formcept.onyx.lsh-onyx.core
```
Comment (anujsrc) [inline clojure/onyx-impl/locality-sensitive-hashing/src/org/formcept/onyx/lsh_onyx/core.clj]: Add FC License Header-

```
;;
;; Copyright (c) 2011-onwards, FORMCEPT [http://www.formcept.com]
;; All rights reserved.
;;
;; NOTICE:  All information contained herein is, and remains
;; the property of FORMCEPT and its suppliers, if any.
;; The intellectual and technical concepts contained
;; herein are proprietary to FORMCEPT and its suppliers and
;; may be covered by U.S. and Foreign Patents, patents in process,
;; and are protected by trade secret or copyright law. Dissemination
;; of this information or reproduction of this material
;; is strictly forbidden unless prior written permission is obtained
;; from FORMCEPT.
;;
```

File: clojure/onyx-impl/onyx-utils/src/org/formcept/onyx/onyx_utils/aggregation/core.cljc [hunk 7]
```diff
@@ -0,0 +1,34 @@
+(ns org.formcept.onyx.onyx-utils.aggregation.core)
```
Comment (anujsrc) [inline clojure/onyx-impl/onyx-utils/src/org/formcept/onyx/onyx_utils/aggregation/core.cljc]: Add FC License Header-

```
;;
;; Copyright (c) 2011-onwards, FORMCEPT [http://www.formcept.com]
;; All rights reserved.
;;
;; NOTICE:  All information contained herein is, and remains
;; the property of FORMCEPT and its suppliers, if any.
;; The intellectual and technical concepts contained
;; herein are proprietary to FORMCEPT and its suppliers and
;; may be covered by U.S. and Foreign Patents, patents in process,
;; and are protected by trade secret or copyright law. Dissemination
;; of this information or reproduction of this material
;; is strictly forbidden unless prior written permission is obtained
;; from FORMCEPT.
;;
```

File: clojure/onyx-impl/onyx-utils/src/org/formcept/onyx/onyx_utils/string_utils/core.clj [hunk 8]
```diff
@@ -0,0 +1,80 @@
+(ns org.formcept.onyx.onyx-utils.string-utils.core
```
Comment (anujsrc) [inline clojure/onyx-impl/onyx-utils/src/org/formcept/onyx/onyx_utils/string_utils/core.clj]: Add FC License Header-

```
;;
;; Copyright (c) 2011-onwards, FORMCEPT [http://www.formcept.com]
;; All rights reserved.
;;
;; NOTICE:  All information contained herein is, and remains
;; the property of FORMCEPT and its suppliers, if any.
;; The intellectual and technical concepts contained
;; herein are proprietary to FORMCEPT and its suppliers and
;; may be covered by U.S. and Foreign Patents, patents in process,
;; and are protected by trade secret or copyright law. Dissemination
;; of this information or reproduction of this material
;; is strictly forbidden unless prior written permission is obtained
;; from FORMCEPT.
;;
```

**Review by anujsrc — CHANGES_REQUESTED** [review]

**Comment by anujsrc** [comment]
@punit-naik - change the commit message and PR title to ``Onyx LSH``

File: clojure/build.sh (hunk 1 above)
Comment (punit-naik) [reply clojure/build.sh]: Done.

File: clojure/onyx-impl/README.md (hunk 2 above)
Comment (punit-naik) [reply clojure/onyx-impl/README.md]: Done.

File: clojure/onyx-impl/build.sh (hunk 3 above)
Comment (punit-naik) [reply clojure/onyx-impl/build.sh]: Done.

File: clojure/onyx-impl/locality-sensitive-hashing/README.md (hunk 4 above)
Comment (punit-naik) [reply clojure/onyx-impl/locality-sensitive-hashing/README.md]: Done.

File: clojure/onyx-impl/onyx-utils/README.md (hunk 5 above)
Comment (punit-naik) [reply clojure/onyx-impl/onyx-utils/README.md]: Done.

File: clojure/onyx-impl/locality-sensitive-hashing/src/org/formcept/onyx/lsh_onyx/core.clj (hunk 6 above)
Comment (punit-naik) [reply clojure/onyx-impl/locality-sensitive-hashing/src/org/formcept/onyx/lsh_onyx/core.clj]: Done, thanks for the input.

File: clojure/onyx-impl/onyx-utils/src/org/formcept/onyx/onyx_utils/aggregation/core.cljc (hunk 7 above)
Comment (punit-naik) [reply clojure/onyx-impl/onyx-utils/src/org/formcept/onyx/onyx_utils/aggregation/core.cljc]: Done, thanks for the input.

File: clojure/onyx-impl/onyx-utils/src/org/formcept/onyx/onyx_utils/string_utils/core.clj (hunk 8 above)
Comment (punit-naik) [reply clojure/onyx-impl/onyx-utils/src/org/formcept/onyx/onyx_utils/string_utils/core.clj]: Done, thanks for the input.

**Comment by punit-naik** [comment]
@anujsrc I have done all the changes you requested.

**Comment by anujsrc** [comment]
@punit-naik - rename the commit to ``Onyx LSH``. Also, fix these warnings-

```
Warning: specified :main without including it in :aot. 
Implicit AOT of :main will be removed in Leiningen 3.0.0. 
If you only need AOT for your uberjar, consider adding :aot :all into your
:uberjar profile instead.
Compiling org.formcept.onyx.lsh-onyx.core
log4j:WARN No appenders could be found for logger (com.amazonaws.AmazonWebServiceClient).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
```

**Comment by anujsrc** [comment]
@punit-naik - please sync with base branch and check the Travis build

**Pull request closed without merging**